pkg/operator/controller/garden/garden/reconciler_reconcile.go

```diff
@@ -638,16 +634,26 @@ func (r *Reconciler) runRuntimeSetupFlow(ctx context.Context, log logr.Logger, g
 			Fn:   component.OpWait(c.istioCRD).Deploy,
 		})
 		deployGardenerResourceManager = g.Add(flow.Task{
-			Name:         "Deploying and waiting for gardener-resource-manager to be healthy",
+			Name:         "Deploying gardener-resource-manager",
 			Fn:           component.OpWait(c.gardenerResourceManager).Deploy,
 			Dependencies: flow.NewTaskIDs(deployEtcdCRD, deployVPACRD, deployIstioCRD),
 		})
+		_ = g.Add(flow.Task{
+			Name:         "Waiting for gardener-resource-manager to be healthy",
+			Fn:           c.gardenerResourceManager.Wait,
+			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager),
+		})
```

Comment:
I do not understand why we split deployment of `gardener-resource-manager` and check of health, but still wait for it in the first step. It feels like `component.OpWait(...)` should have been removed in the first step. Otherwise the split seems useless.

Reply:
Good catch, thank you!